godot-cell/src/blocking_cell.rs

```diff
@@ -54,10 +54,18 @@ impl<T> GdCellBlocking<T> {
             tracker_guard = self.block_immut(tracker_guard);
         }
 
+        let should_claim_mut = !self.is_currently_bound();
+
         let inner_guard = self.inner.as_ref().borrow()?;
 
         tracker_guard.increment_current_thread_shared_count();
 
+        // Claiming the mutable reference for the current thread, if the cell is currently not bound, prevents the thread that initialized the
+        // cell from panicking when it tries to borrow the cell mutably while shared references exist.
```

Comment:
This is a very long sentence loaded with interactions.

Could you maybe outline a scenario that would happen if this check _didn't_ exist? Feel free to use multiple lines and mention each step chronologically.

Reply:
I rewrote the comment. I hope it's much clearer now.